tidalapi/request.py

```diff
@@ -60,11 +60,24 @@ def basic_request(self, method, path, params=None, data=None, headers=None):
         request = self.session.request_session.request(method, url, params=request_params, data=data, headers=headers)
 
         refresh_token = self.session.refresh_token
-        if not request.ok and request.json()['userMessage'].startswith("The token has expired.") and refresh_token:
-            log.debug("The access token has expired, trying to refresh it.")
-            refreshed = self.session.token_refresh(refresh_token)
-            if refreshed:
-                request = self.basic_request(method, url, params, data, headers)
+        if not request.ok and refresh_token:
+            json_resp = None
+            try:
+                json_resp = request.json()
+            except requests.JSONDecodeError:
+                pass
+
+            if (
+                json_resp and
+                json_resp.get('userMessage', '').startswith("The token has expired.")
+            ):
+                log.debug("The access token has expired, trying to refresh it.")
+                refreshed = self.session.token_refresh(refresh_token)
+                if refreshed:
+                    request = self.basic_request(method, url, params, data, headers)
+            else:
+                log.warning('HTTP error on %s %s: %d', request.status_code)
```

Comment:
The formatting seems a bit off here, are the %s parts supposed to be line numbers?

Reply:
Just removed the %s parts, let me know if you intended something else